guardian/media-atom-maker
# Replace react-drag-sortable with dnd-kit

## What does this change?

This PR replaces [`react-drag-sortable`,](https://www.npmjs.com/package/react-drag-sortable) which was used to implement the drag-and-drop tag list, with `dnd-kit`.

`react-drag-sortable` is no longer maintained and contains a high-severity vulnerability.

It also makes some changes to `eslint` in order to support the new `.tsx` file.

(Prequel PR: https://github.com/guardian/media-atom-maker/pull/1094)

## How to test

1. Deploy to CODE (saving tags doesn't work properly with the current local config setup).
2. Open a media atom (i.e. video)
3. Add multiple tags to the Composer Keywords list
4. Rearrange them by drag and drop
5. Close one or two
6. Save your changes

Are the tags saved as expected?

## Images
![dragndrop](https://user-images.githubusercontent.com/34686302/217558690-7d73fc95-5494-4e09-b3f4-727bfb41a0fa.gif)

As a bonus, the new drag and drop interface is keyboard-accessible. Select focused tags with <kbd>Enter</kbd>, move with the <kbd>up</kbd>/<kbd>down</kbd> arrow keys and select location with <kbd>Enter</kbd> or <kbd>Esc</kbd>.

![dragndropkeyboard](https://user-images.githubusercontent.com/34686302/217565144-465d63c2-5aa1-4fe7-8c76-752953cbe4c0.gif)

Moving tags can be accomplished with VoiceOver too, though upon closing a tag VoiceOver seems to lose its place in the document. As far as I'm aware, no `media-atom-maker` user accesses it via a screenreader so I wouldn't consider  fixing this to be a priority.

## Accessibility
